src/styles/main.scss

```diff
@@ -1,3 +1,54 @@
 body {
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+@import "variables";
+
+@for $i from 28 through 31 {
+  .calendar.calendar--month-length--#{$i}
+  .calendar__day:nth-child(n + #{$i + 1}) {
+    display: none;
+  }
+}
+
+@for $i from 1 through 31 {
+  .calendar__day:nth-child(n + #{$i})::before {
+    content: "#{$i}";
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
```

Comment:
Better leave only content here. You can set all other properties just under the `calendar__day` selector, no need to use loop for that. Btw, you can center the text using `text-align` and `line-height` properties